### PR TITLE
Usage Examples Feature Branch

### PR DESCRIPTION
**Pull request by mongodben**
# Pull Request Info

Feature branch to aggregate usage examples before merging into `master`.

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - N/A
Staging - https://docs-mongodbcom-staging.corp.mongodb.com/kotlin/docsworker-xlarge/usage_examples/

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [ ] Are all the links working?

### CODE REVIEW

File: source/usage-examples/find.txt [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+returned documents.
+
+For more information on the ``sort()`` method, see our
+:doc:`guide on Sorting </fundamentals/crud/read-operations/sort/>`.
+For more information on the ``projection()`` method, see our
+:doc:`guide on Projections </fundamentals/crud/read-operations/project/>`
+
+The ``find()`` method returns an instance of ``FindFlow``, a class
+that offers several methods to access, organize, and traverse the results.
+.. TODO (DOCSP-29176) Update with accurate inheritance information when available
```
Comment (cbullinger) [inline source/usage-examples/find.txt:30]:
```suggestion

.. TODO (DOCSP-29176) Update with accurate inheritance information when available
```

File: source/usage-examples/findOne.txt [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+``projection()`` which configures the fields included in the returned documents.
+
+For more information on the ``sort()`` method, see our
+:doc:`guide on Sorting </fundamentals/crud/read-operations/sort/>`.
+For more information on the ``projection()`` method, see our
+:doc:`guide on Projections </fundamentals/crud/read-operations/project/>`
+
+The ``find()`` method returns an instance of ``FindFlow``, a class
+that offers several methods to access, organize, and traverse the results.
+.. TODO (DOCSP-29176) Update with accurate inheritance information when available
```
Comment (cbullinger) [inline source/usage-examples/findOne.txt:29]:
```suggestion

.. TODO (DOCSP-29176) Update with accurate inheritance information when available
```

**Review by cbullinger — APPROVED** [review]

File: source/usage-examples/findOne.txt [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+.. ``FindFlow`` also inherits methods from its parent class ``Flow`` from the
+.. Kotlin Coroutines library, such as ``first()`` and ``firstOrNull()``.
```
Comment (rustagir) [inline source/usage-examples/findOne.txt:31]: These TODOs are showing up in staging

File: source/usage-examples/findOne.txt [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+Example
+-------
+
+The following snippet finds a single document from the ``movies`` collection.
+It uses the following objects and methods:
+
+- A **query filter** that is passed to the ``find()`` method. The ``eq``
+  filter matches only movies with the title exactly matching the text
+  ``'The Room'``.
```
Comment (rustagir) [inline source/usage-examples/findOne.txt:50]:
```suggestion
  ``"The Room"``.
```

File: source/usage-examples/find.txt [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+.. ``FindFlow`` also inherits methods from its parent class ``Flow`` from the
+.. Kotlin Coroutines library.
```
Comment (rustagir) [inline source/usage-examples/find.txt:32]: Still showing up as TODOs

File: source/usage-examples/find.txt [hunk 6]
```diff
@@ -0,0 +1,70 @@
+.. _kotlin-usage-find:
+
+=======================
+Find Multiple Documents
+=======================
+
+.. default-domain:: mongodb
```
Comment (rustagir) [inline source/usage-examples/find.txt:7]: This setting doesn't do anything anymore, so we can remove from all files (not in scope but just noting)

File: source/examples/generated/FindOne.snippet.find-one-usage-example.kt [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    val database = mongoClient.getDatabase("sample_mflix")
+    val collection = database.getCollection<Movie>("movies")
+
+    val projectionFields= Projections.fields(
+        Projections.include(Movie::title.name, Movie::imdb.name),
+        Projections.excludeId()
+    )
+    val resultsFlow = collection.find(lt(Movie::title.name, "The Room"))
+        .projection(projectionFields)
+        .sort(Sorts.descending("${Movie::imdb.name}.${Movie.IMDB::rating.name}"))
```
Comment (rustagir) [inline source/examples/generated/FindOne.snippet.find-one-usage-example.kt:25]: The parameters for the Sort field look different here than in the FInd example (inclusion of $) -- do they both work? Maybe we should standardize

File: source/usage-examples/updateOne.txt [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+
+For more information on the types of exceptions raised under specific
+conditions, see the ``updateOne()`` API documentation linked at the
+bottom of this page.
+
+Example
+-------
+
+In this example, we perform the following updates to the first match for our query 
+in the ``movies`` collection of the ``sample_mflix`` database:
```
Comment (rustagir) [inline source/usage-examples/updateOne.txt:60]: The page doesn't describe the query

File: source/usage-examples/updateMany.txt [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+For more information on the types of exceptions raised under specific
+conditions, see the API documentation for ``updateMany()``, linked at the
+bottom of this page.
+
+Example
+-------
+
+In this example, we update documents that match our query in the ``movies``
+collection of the ``sample_mflix`` database. We perform the following
+updates to the matching documents:
```
Comment (rustagir) [inline source/usage-examples/updateMany.txt:63]: The page doesn't describe the query

File: source/usage-examples/deleteMany.txt [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+this page.
+
+Example
+-------
+
+The following snippet deletes multiple documents from the ``movies``
+collection in the ``sample_mflix`` database. 
+
+The query filter passed to the ``deleteMany()`` method matches all
+movie documents that contain a ``rating`` of less than "2.9" in the ``imdb``
```
Comment (rustagir) [inline source/usage-examples/deleteMany.txt:35]:
```suggestion
movie documents that contain a ``rating`` of less than ``2.9`` in the ``imdb``
```

File: source/usage-examples/findOne.txt (hunk 4 above)
Comment (rustagir) [reply source/usage-examples/findOne.txt:50]: fix elsewhere too if possible to match the way strings appear in the code examples

File: source/usage-examples/findOne.txt (hunk 2 above)
Comment (cbullinger) [inline source/usage-examples/findOne.txt:29]:
```suggestion

.. TODO (DOCSP-29176) Update with accurate inheritance information when available
```

File: source/usage-examples/find.txt (hunk 1 above)
Comment (cbullinger) [inline source/usage-examples/find.txt:30]:
```suggestion

.. TODO (DOCSP-29176) Update with accurate inheritance information when available
```

File: source/examples/generated/FindOne.snippet.find-one-usage-example.kt (hunk 7 above)
Comment (cbullinger) [reply source/examples/generated/FindOne.snippet.find-one-usage-example.kt:25]: They both work. The inclusion of $ is due to the dot notation with the property references to access the imdb.rating field

**Pull request merged**